### ISSUE TITLE
fermionic split-CTM: split vs shim energies disagree once bra is non-trivial

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#392]
## Summary

After fixing #391 (canonicalising the interlayer-bond charges so the ket and bra halves of each split-T edge carry identical raw charges), the existing fermionic parity test `tests/test_split_ctm_tensor.py::TestSplitRDMsFermionic::test_fermionic_energy_matches_shim` reveals a pre-existing discrepancy between the two energy code paths.

Reference run on `make_random_fermionic_site(D, d=2, seed=70)` after 20 CTM sweeps:

| (D, chi) | E_split | E_shim |
|----------|---------|--------|
| (2, 6)   | 1.966879 | 1.363992 |
| (2, 8)   | -0.542403 | 2.514577 |
| (3, 12)  | 4.396583 | -5.666788 |

`compute_energy_split_ctm_tensor` (split-aware path through `_rdm{2x1,1x2}_split_tensor` + `A.bar_super()`) and `compute_energy_ctm_tensor` (standard path on `_split_env_to_tensor_standard(env)` via `_build_double_layer_open_tensor`) compute the same physical observable from the same converged env, so they should agree to machine precision (the U(1)-trivial case in `TestSplitRDMs` does).

## Why this was hidden

Pre-#391, the bra's identity-like init was *vacuous* on `FermionParity` sites because the SVD-bond raw charges between ket and bra mismatched: the bra-side blocks that the iteration would have populated lived in sectors the relabel-and-contract step couldn't reach (raw `qI=2` vs `qI=-1`), so the bra stayed at zero. With both energies at `0.0`, the parity assertion `jnp.allclose(E_split, E_shim, atol=1e-10)` passed trivially. Pre-fix at `(D=2, chi=6)`:

```
PRE-FIX D=2 chi=6: E_split=0.000000, E_shim=0.000000
PRE-FIX D=3 chi=12: E_split=0.000000, E_shim=0.000000
```

Once #391 lets the bra populate, both paths produce non-trivial values that disagree.

## Likely culprits

- A Koszul / bar-super sign convention in `_rdm{2x1,1x2}_split_tensor` that doesn't match what `_build_double_layer_open_tensor` bakes into the standard env.
- A flow-direction mismatch on the merged-T legs: the bra's chi-leg flow in `_EDGE_BRA_SPECS` differs from what `_split_env_to_tensor_standard` produces (`_ensure_*_flows` is a no-op).
- An inversion-pair / leg-order accounting difference between the split contraction order and the standard `ao_A · merged_T` order.

## Suggested first steps

1. Add a Tier-1 (bosonic, U(1)) sanity test asserting `E_split == E_shim` at `(D, chi)` where the bra is genuinely populated to confirm the U(1) baseline still matches post-#391.
2. Build a minimal reproduction with `D=2`, `chi=4` and **all-zero bra** vs **non-trivial bra** to localise where the two paths diverge.
3. Compare against a third reference: the standard CTM path run from scratch (no shim) on the same `A`, to determine which of `E_split` and `E_shim` (if either) is correct.

The Tier-2 parity test will be scoped down in #391 with a docstring pointer to this issue.

**Comment by yingjerkao** [comment]
PR #394 ships a pragmatic fix: `compute_energy_split_ctm_tensor*` now detects fermionic input via `_is_fermionic_site(A)` and falls back to the shim path (`_split_env_to_tensor_standard` + `compute_energy_ctm_tensor*`). The Tier-2 parity test now passes (no more `xfail(strict=True)`).

The split-aware χ²·D⁴ path remains unfixed for fermions. Three rounds of probes (replacing `bar_super()` with `bar()` in the energy alone, in convergence + energy, and inserting the corrective phase per-block) all failed to make split == shim. Recovering the χ²·D⁴ benefit for fermions remains open and tracks under this same issue — please leave it open until a deeper Koszul-aware fix lands. Notes on what didn't work and what to try next are in the PR description.
